go/service/gregor.go

```diff
@@ -55,14 +56,65 @@ func (g *gregorHandler) OnDoCommandError(err error, nextTime time.Duration) {
 }
 
 func (g *gregorHandler) ShouldRetry(name string, err error) bool {
+	g.G().Log.Debug("gregor handler: should retry: name %s, err %v (returning false)", name, err)
 	return false
 }
 
 func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
-	return false
+	if err == nil {
+		return false
+	}
+
+	g.G().Log.Debug("gregor handler: should retry on connect, err %v (returning true)", err)
+	return true
 }
 
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
 	return nil
 }
+
+func (g *gregorHandler) OnLogin() {
+	g.G().Log.Debug("gregor handler: OnLogin")
+	if err := g.tryAuth(context.Background()); err != nil {
+		g.G().Log.Debug("gregor handler: OnLogin tryAuth error: %s", err)
```

Comment:
Should this hook (I assume which comes from the interface in the RPC library) take a return `error` parameter?